djangorestframework_camel_case/util.py

```diff
@@ -48,21 +54,13 @@ def camelize(data, **options):
             else:
                 new_dict[new_key] = result
         return new_dict
-    if is_iterable(data) and not isinstance(data, str):
+    if isinstance(data, Iterable) and not isinstance(data, str):
         return [camelize(item, **options) for item in data]
     return data
 
 
-def get_underscoreize_re(options):
-    if options.get("no_underscore_before_number"):
-        pattern = r"([a-z0-9]|[A-Z]?(?=[A-Z](?=[a-z])))([A-Z])"
-    else:
-        pattern = r"([a-z0-9]|[A-Z]?(?=[A-Z0-9](?=[a-z0-9]|(?<![A-Z])$)))([A-Z]|(?<=[a-z])[0-9](?=[0-9A-Z]|$)|(?<=[A-Z])[0-9](?=[0-9]|$))"
-    return re.compile(pattern)
```

Comment:
About a 10% improvement for the first pattern, and 27% for the second, at least on my machine.
The change I made allows the pattern to be compiled exactly once on module instantiation, rather than repeatedly every time this function is entered.

Before:
```python
from djangorestframework_camel_case.util import camel_to_underscore
timeit.timeit(lambda: camel_to_underscore("someName"))
2.591615671000909
timeit.timeit(lambda: camel_to_underscore("someName5", some_number_before_number=True))
2.839183228003094
```

```python
import timeit
from djangorestframework_camel_case.util import camel_to_underscore
timeit.timeit(lambda: camel_to_underscore("someName"))
2.3492436910019023
timeit.timeit(lambda: camel_to_underscore("someName5", some_number_before_number=True))
2.227818359999219
```